dashboard/tasks_test.py

```diff
@@ -44,26 +43,96 @@ def setUpTestData(cls):
 
     def setUp(self):
         super(TasksTest, self).setUp()
-        self.students = [self.user.id, self.user2.id]
+        self.students = [self.user1.id, self.user2.id]
 
-    @override_settings(CELERY_ALWAYS_EAGER=True)
     def test_celery_task_works(self):
         """
         Assert task schedule using celery beat.
         """
         self.assertTrue(batch_update_user_data.delay())
 
-    @override_settings(CELERY_ALWAYS_EAGER=True)
     @mock.patch('dashboard.api_edx_cache.CachedEdxDataApi.update_cache_if_expired', new_callable=mock.MagicMock)
     @mock.patch('backends.utils.refresh_user_token', autospec=True)
-    def test_student_enrollments_called_task(
-            self, mocked_refresh, mocked_refresh_cache):
+    def test_student_enrollments_called_task(self, mocked_refresh, mocked_refresh_cache):
         """
-        Assert get_student_enrollments is actually called
+        Assert get_student_enrollments is actually called in happy path
         """
         batch_update_user_data_subtasks.s(self.students).apply(args=()).get()
         assert mocked_refresh_cache.call_count == len(self.all_users) * len(CachedEdxDataApi.SUPPORTED_CACHES)
         assert mocked_refresh.call_count == len(self.all_users)
         for user, cache_type in product(self.all_users, CachedEdxDataApi.SUPPORTED_CACHES):
             mocked_refresh_cache.assert_any_call(user, mock.ANY, cache_type)
             mocked_refresh.assert_any_call(user.social_auth.get(provider=EdxOrgOAuth2.name))
```

Comment:
Can you add a test where the first user has a failure somewhere but the second user doesn't, and therefore the mocked methods are called only for the second user?